imap_processing/codice/decompress.py

```diff
@@ -26,115 +26,117 @@
+def _apply_lossy_b(compressed_bytes: bytes) -> list[int]:
     """
-    Apply 8-bit to 32-bit Lossy B decompression algorithm.
+    Apply 8-bit to 32-bit Lossy A decompression algorithm.
 
     The Lossy B algorithm uses a lookup table imported into this module.
 
     Parameters
     ----------
-    compressed_value : int
-        The compressed 8-bit value.
+    compressed_bytes : bytes
+        The compressed byte stream.
 
     Returns
     -------
-    int
-        The 24- or 32-bit decompressed value.
+    decompressed_values : list[int]
+        The 24- or 32-bit decompressed values.
     """
-    return LOSSY_B_TABLE[compressed_value]
+    compressed_values = list(compressed_bytes)
+    decompressed_values = [LOSSY_B_TABLE[item] for item in compressed_values]
+    return decompressed_values
 
 
-def _apply_lzma_lossless(compressed_value: Union[int, bytes]) -> int:
+def _apply_lzma_lossless(compressed_bytes: bytes) -> bytes:
     """
     Apply LZMA lossless decompression algorithm.
 
     Parameters
     ----------
-    compressed_value : int or bytes
-        The compressed 8-bit value.
+    compressed_bytes : bytes
+        The compressed byte stream.
 
     Returns
     -------
-    decompressed_value : int
-        The 24- or 32-bit decompressed value.
+    lzma_decompressed_values : bytes
+        The 24- or 32-bit lzma decompressed values.
     """
-    if isinstance(compressed_value, int):
-        bytes_compressed_value = compressed_value.to_bytes(compressed_value, "big")
-    else:
-        bytes_compressed_value = compressed_value
-    decompressed_value = lzma.decompress(bytes_compressed_value)
-    decompressed_value_int = int.from_bytes(decompressed_value, byteorder="big")
+    lzma_decompressed_values = lzma.decompress(compressed_bytes)
 
-    return decompressed_value_int
+    return lzma_decompressed_values
 
 
-def decompress(compressed_value: int, algorithm: IntEnum) -> int:
+def decompress(compressed_binary: str, algorithm: IntEnum) -> list[int]:
     """
-    Will decompress the value.
+    Perform decompression on a binary string into a list of integers.
 
     Apply the appropriate decompression algorithm(s) based on the value
     of the ``algorithm`` attribute. One or more individual algorithms may be
     applied to a given compressed value.
 
     Parameters
     ----------
-    compressed_value : int
-        The 8-bit compressed value to decompress.
+    compressed_binary : str
+        The compressed binary string.
     algorithm : int
         The algorithm to apply. Supported algorithms are provided in the
         ``codice_utils.CoDICECompression`` class.
 
     Returns
     -------
-    decompressed_value : int
-        The 24- or 32-bit decompressed value.
+    decompressed_values : list[int]
+        The 24- or 32-bit decompressed values.
     """
+    # Convert the binary string to a byte stream
+    compressed_bytes = bitarray.bitarray(compressed_binary).tobytes()
```

Comment:
Does this work instead?

```suggestion
    compressed_bytes = int(compressed_binary, 2).to_bytes((len(compressed_binary) + 7) // 8, byteorder='big')
```

Reply:
I'm trying to avoid bitarray/bitstring if possible as I think you're a bit lucky here because it is brought in via space_packet_parser, but that might be removed in the future (I hope): https://github.com/medley56/space_packet_parser/pull/51

Reply:
Yes, that works! I'll update that.